PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx

```diff
@@ -728,46 +987,144 @@ struct meanPtFlucId {
+        }
+        if (twopart_Ka != 0) {
+          hist.fill(HIST(dire[2]) + HIST("h_twopart_Mult_var"), NTPC, twopart_Ka, N_FT0C);
+        }
+      }
+      if (N_Ka > 2) {
+        if (mean_Q1_Ka != 0)
+          hist.fill(HIST(dire[2]) + HIST("h_mean_Q1_Mult_skew"), NTPC, mean_Q1_Ka, N_FT0C);
+        if (twopart_Ka != 0)
+          hist.fill(HIST(dire[2]) + HIST("h_twopart_Mult_skew"), NTPC, twopart_Ka, N_FT0C);
+        if (threepart_Ka != 0)
+          hist.fill(HIST(dire[2]) + HIST("h_threepart_Mult_skew"), NTPC, threepart_Ka, N_FT0C);
+      }
+
+      if (N_Ka > 3) {
+        if (mean_Q1_Ka != 0)
+          hist.fill(HIST(dire[2]) + HIST("h_mean_Q1_Mult_kurto"), NTPC, mean_Q1_Ka, N_FT0C);
+        if (twopart_Ka != 0)
+          hist.fill(HIST(dire[2]) + HIST("h_twopart_Mult_kurto"), NTPC, twopart_Ka, N_FT0C);
+        if (threepart_Ka != 0)
+          hist.fill(HIST(dire[2]) + HIST("h_threepart_Mult_kurto"), NTPC, threepart_Ka, N_FT0C);
+        if (fourpart_Ka != 0)
+          hist.fill(HIST(dire[2]) + HIST("h_fourpart_Mult_kurto"), NTPC, fourpart_Ka, N_FT0C);
+      }
+
+      parts(Q1_Pr, Q2_Pr, Q3_Pr, Q4_Pr, N_Pr, &mean_Q1_Pr, &twopart_Pr, &threepart_Pr, &fourpart_Pr);
+      if (N_Pr > 1) {
+        hist.fill(HIST(dire[3]) + HIST("p_CheckNCH"), NTPC, (1 / std::sqrt(1 - (1 / static_cast<double>(N_Pr)))));
+        hist.fill(HIST(dire[3]) + HIST("h_CheckNCH"), NTPC, (1 / std::sqrt(1 - (1 / static_cast<double>(N_Pr)))), N_FT0C);
+        if (mean_Q1_Pr != 0) {
+          hist.fill(HIST(dire[3]) + HIST("h_mean_Q1"), mean_Q1_Pr);
+          hist.fill(HIST(dire[3]) + HIST("p_mean_Q1_Mult_var"), NTPC, mean_Q1_Pr);
+          hist.fill(HIST(dire[3]) + HIST("h_mean_Q1_Mult_var"), NTPC, mean_Q1_Pr, N_FT0C);
+        }
+        if (twopart_Pr != 0) {
+          hist.fill(HIST(dire[3]) + HIST("h_twopart_Mult_var"), NTPC, twopart_Pr, N_FT0C);
+        }
+      }
+
+      if (N_Pr > 2) {
+        if (mean_Q1_Pr != 0)
+          hist.fill(HIST(dire[3]) + HIST("h_mean_Q1_Mult_skew"), NTPC, mean_Q1_Pr, N_FT0C);
+        if (twopart_Pr != 0)
+          hist.fill(HIST(dire[3]) + HIST("h_twopart_Mult_skew"), NTPC, twopart_Pr, N_FT0C);
+        if (threepart_Pr != 0)
+          hist.fill(HIST(dire[3]) + HIST("h_threepart_Mult_skew"), NTPC, threepart_Pr, N_FT0C);
+      }
+
+      if (N_Pr > 3) {
+        if (mean_Q1_Pr != 0)
+          hist.fill(HIST(dire[3]) + HIST("h_mean_Q1_Mult_kurto"), NTPC, mean_Q1_Pr, N_FT0C);
+        if (twopart_Pr != 0)
+          hist.fill(HIST(dire[3]) + HIST("h_twopart_Mult_kurto"), NTPC, twopart_Pr, N_FT0C);
+        if (threepart_Pr != 0)
+          hist.fill(HIST(dire[3]) + HIST("h_threepart_Mult_kurto"), NTPC, threepart_Pr, N_FT0C);
+        if (fourpart_Pr != 0)
+          hist.fill(HIST(dire[3]) + HIST("h_fourpart_Mult_kurto"), NTPC, fourpart_Pr, N_FT0C);
       }
-      if (twopart_Pr != 0)
-        hist.fill(HIST(dire[3]) + HIST("h_twopart_Mult_var"), NTPC, twopart_Pr);
     }
   }
```

Comment:
I was trying to use templates to reduce this histograms replication for different Identified particles in the code. but it wasn't working.